### PR TITLE
Refactor mas styles to leverage consonant for CTAs

### PR DESCRIPTION
**Pull request by yesil**
remove spectrum variable dependencies

with some exceptions.

Resolves: [MWPW-163621](https://jira.corp.adobe.com/browse/MWPW-163621)

**Test URLs:**
- Before: https://main--milo--adobecom.aem.live/libs/features/mas/docs/ccd.html
- After: https://mwpw-163621--milo--adobecom.aem.live/libs/features/mas/docs/ccd.html

### CODE REVIEW

File: libs/features/mas/dist/mas.css [hunk 1]
```diff
@@ -0,0 +1,170 @@
+/* Colors */
+merch-card {
+  --link-color: #035fe6;
+  --link-hover-color: #136ff6;
+  --link-color-dark: #0265dc;
+  --link-hover-color-dark: #0265dc;
+  --background-color: #fff;
+  --text-color: #2c2c2c;
+  --color-white: #FFF;
```
Comment (narcis-radu) [inline libs/features/mas/dist/mas.css:9]: nit: consistency for hex values
```suggestion
  --color-white: #fff;
```

File: libs/features/mas/dist/mas.css [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  color: var(--color-white);
+}
+
+.light .con-button {
+  border-color: var(--text-color);
+  color: var(--text-color);
+}
+
+.light .con-button:hover,
+.light .con-button:active {
```
Comment (narcis-radu) [inline libs/features/mas/dist/mas.css:90]: I don't see a `.dark .con-button:active`

File: libs/features/mas/dist/mas.css [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+.con-button.fill[aria-disabled="true"] {
+  background-color: transparent;
+  border-color: var(--color-gray-500);
+  color: var(--color-gray-500);
+}
+
+.con-button.blue[aria-disabled="true"],
+.con-button.fill[aria-disabled="true"] {
+  background-color: var(--color-gray-500);
+  color: var(--color-gray-300);
```
Comment (narcis-radu) [inline libs/features/mas/dist/mas.css:152]: Is this any different from https://github.com/adobecom/milo/pull/3306/files#diff-ed23b36063b3814e2ce691dcf650d3aefaf5c9fa9d7a9d8336113140768e261dR146 ? Can we re-arrange styles so the code is not duplicated?

File: libs/features/mas/src/hydrate.js [hunk 4]
```diff
@@ -186,28 +156,67 @@ function createSpectrumButton(cta, strong, aemFragmentMapping, cardVariant) {
...
+    if (treatment === 'outline') {
+        cta.classList.add('outline');
+    }
+    if (variant === 'accent') {
         cta.classList.add('blue');
+    } else if (variant === 'black') {
+        cta.classList.add('fill');
+    }
+    const size = aemFragmentMapping.ctas.size ?? 'm';
+    if (size === 's') {
```
Comment (narcis-radu) [inline libs/features/mas/src/hydrate.js:170]: nit: this can be written inline
```suggestion
    if (size === 's') cta.classList.add('button-s');
```

**Comment by afmicka** [comment]
@yesil Seeing the following issues:

1. slice card widths are not correct, the cards are smaller on ccd page with the width of 305px or 307.2 etc, ccd page has them on max width of 322px
2. Background color is changed. Should be light grey for light mode (specifically asked i think) rgb (248,248,248), now it is white rgb(255,255,255). Also is different for the dark mode: should be rgb(70,70,70) but here is rgb(30,30,30)
3. border color is changed for bot light and dark mode:
- light is now rgb(234,234,234), was and should be rgb(230,230,230)
- dark is now rgb(63,63,63), was and should be rgb(70,70,70)
4. Title color mismatch:
- suggested light: was rgb(34,34,34), now rgb(36,36,36)
5. Description text color mismatch:
- suggested card light: was and should be gray rgb(70,70,70), now rgb(36,36,36)
- slice card light: was rgb(34,34,34), rgb(36,36,36)
6. Price color mismatch for all cards the same as no.4
7. CTA color:
- suggested dark: was rgb(235, 235, 235), now full white rgb(255,255,255)
- slice card dark: was and should be rgb((3, 103, 224), now rgb(2, 101, 220). Note that CTA color should not be the same for light and dark mode, you made it the same now. The value you have here is correct for the light theme but not for the dark
8. CTA highlight dark mode - was transparent grey, not solid white
Screenshot 2024-12-04 at 15 44 56
Screenshot 2024-12-04 at 15 44 52

9. See terms link color:
- suggested:  was rgb(20, 122, 243), now rgb(20, 115, 230)
- slice: the same as no.4

File: libs/features/mas/src/hydrate.js [hunk 5]
```diff
@@ -218,19 +227,27 @@ export function processCTAs(fields, merchCard, aemFragmentMapping, variant) {
...
-      merchCard.setAttribute(ANALYTICS_SECTION_ATTR, cardAnalyticsId);
-      merchCard.querySelectorAll(`a[data-analytics-id]`).forEach((link, index) => {
-        link.setAttribute(ANALYTICS_LINK_ATTR, `${link.dataset.analyticsId}-${index + 1}`);
-      });
+    const { tags } = fields;
+    const cardAnalyticsId = tags
+        ?.find((tag) => tag.startsWith(ANALYTICS_TAG))
+        ?.split('/')
+        .pop();
+    if (cardAnalyticsId) {
```
Comment (narcis-radu) [inline libs/features/mas/src/hydrate.js:235]: shouldn't we fail fast here if there's no `cardAnalyticsId` ?

**Comment by yesil** [comment]
@narcis-radu thanks for the review, just got a feedback from CCD team and I can close this one and continue https://github.com/adobecom/milo/pull/3289 which is a similar fix with Spectrum CSS.

**Pull request closed without merging**